### ISSUE TITLE
[12.0] runboat/build fail

### ISSUE THREAD

**Issue by maisim** [OCA/server-tools#2296]
runboat/build fail on PR check on branch 12.0

## Module
database_cleanup , but should be the same for others

## Describe the bug

On build, server-tools/company_country module complain says COUNTRY environment variable miss

`
File "/mnt/data/odoo-addons-dir/setup/company_country/odoo/addons/company_country/models/res_config.py", line 39, in load_company_country 'COUNTRY environment variable with country code is not ' odoo.exceptions.ValidationError: ('COUNTRY environment variable with country code is not set and no localization module is marked to be installed.', None)
`
https://github.com/OCA/server-tools/blob/311142a8aab02bb9f6a2fee9fa0d01d539bacaee/company_country/models/res_config.py#L28-L41

from https://runboat.odoo-community.org/api/v1/builds/b36d577c1-c309-42aa-b1b0-e972138d5871/init-log
generated from this issue : https://github.com/OCA/server-tools/pull/2295

But the environment variable seems to be set correctly in travis config: 
https://github.com/OCA/server-tools/blob/311142a8aab02bb9f6a2fee9fa0d01d539bacaee/.travis.yml#L17

## To Reproduce

submit a PR on 12.0 branchof server-tools repo

**Comment by maisim** [comment]
@sbidoul @gurneyalex

Salut, 
You seem busy, can I do something to get it fixed?

**Comment by sbidoul** [comment]
I suggest settings a default value in the company_country module if the COUNTRY env var is not set.

**Comment by maisim** [comment]
@sbidoul 
Ok I will do that

**Comment by maisim** [comment]
after looking at it more closely, I think that would render the module useless. [company_country](https://github.com/OCA/server-tools/tree/311142a8aab02bb9f6a2fee9fa0d01d539bacaee/company_country)
Its purpose is to force the user to define the country, before the account module is installed so that the correct parameters are set up (l10n*)
If we define a default value, it no longer works.

**Comment by sbidoul** [comment]
We can do `country_code = os.environ.get('COUNTRY', 'US')`. If the user forgot to set the env var, well, they have to start again, but the module still serves its purpose, right?

I don't see any other solution to support GitHub actions and runboat at this stage. TBH I'm trying to see if we can eradicate per-project specific CI configs, and so far we can.

**Comment by maisim** [comment]
When I read the code I feel the goal is really to raise an exception if COUNTRY env var id not defined. (but maybe I'm wrong)

Hi @moylop260 @luisg123v,
As contributors on this addon, do you have an opinion on how to keep the addon working has expected and, in the same time, remove requierement of specific CI config?

**Comment by moylop260** [comment]
> @maisim When I read the code I feel the goal is really to raise an exception if COUNTRY env var id not defined.

Yes, It is correct

It helps to avoid using a default l10n module as Odoo do it

If we use a default l10n module so it is the same than don't use the module itself (original Odoo way)

> @sbidoul 
> I'm trying to see if we can eradicate per-project specific CI configs...

I think we could discuss the following possible solutions

1. Define a global environment variable `COUNTRY=US` in the Dockerfile of the container used for the CI
2. Define a validation in the module itself if the [CI environment variable is defined as true](https://docs.github.com/es/actions/learn-github-actions/environment-variables#default-environment-variables) almost of the CIs is using this variable
e.g.
```python
if not country_code and os.environ.get("CI", '').lower() == 'true':
    country_code = 'US' 
```
4. Define a `variables.sh` file in the project in order to define custom environment variables by project. So, call `source variables.sh` before to run `odoo-bin`, so it could be useful to define `queue channel` and test functionally the queues in runboat for projects using this feature
5. Ignore this module from runboat

What option is better for you?

**Comment by maisim** [comment]
I think 4 is the clean way to manage this case
Being able to customize the environment seems useful to me. Has you suggest, It make sense to do it at project level

**Comment by sbidoul** [comment]
If you really can't have default values, 2/ is the best short term solution. In the longer term we can work on a way to customize oca-ci, and runboat, and the repo template, but that's quite a bit more work to make it cleanly.

**Comment by maisim** [comment]
> If you really can't have default values, 2/ is the best short term solution.

If everyboby is ok I will apply 2
It's on 12.0 I think it's not really bad to have this kind of workaround

> In the longer term we can work on a way to customize oca-ci, and runboat, and the repo template, but that's quite a bit more work to make it cleanly.

@sbidoul  I sould work on this in a few weeks,  with your help, this can be an opportunity for me to better understand OCA's dev workflow

**Comment by sbidoul** [comment]
> I sould work on this in a few weeks, with your help, this can be an opportunity for me to better understand OCA's dev workflow

@maisim cool. Let's first make sure it's necessary, though, because it is non-trivial work. So far it is just one case, for a module that is not being used in 13+. To discuss this, the best is probably to collect the use cases for additional CI and runboat customization. The best place to do this would be a new issue in oca/oca-addons-repo-template.

**Comment by luisg123v** [comment]
> If everyboby is ok I will apply 2

@maisim I agree. Even though 4 is an ideal solution, 2 is the immediate one.

**Comment by thomaspaulb** [comment]
I've meanwhile applied option 2 and this is merged into 12.0 branch.
